### PR TITLE
feat(components-library): Add active filters component

### PR DESCRIPTION
**Pull request by connoratrug**
- Move over ActiveFilters component form filers repo
- Move setup code to demo-data folder and use in both examples and unit tests
- Add  missing build export for DataTimeFilter component

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [ ] Conventional commits (squash if needed)
- [x] No warnings during install
- [x] Updated javascript typing

### CODE REVIEW

**Comment by connoratrug** [comment]
[#7576 Add active filters component](https://trello.com/c/5v4aHFoA/296-7576-add-active-filters-component)

File: packages/components-library/src/components/filters/ActiveFilters.vue [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+### Usage
+```jsx
+
+const selections = {
+  search: 'test',
+  name: 'John',
+  age: [32, 45],
+  fruit: 'orange',
+  color: ['red', 'green']
+}
```
Comment (jvanveen) [inline packages/components-library/src/components/filters/ActiveFilters.vue:182]: Are you sure this covers all use-cases of the component? It seems there is also something related to datetime?

File: packages/components-library/tests/demo-data/filterMocks.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  { text: 'Mango', value: 'mango' },
+  { text: 'Breadfruit', value: 'breadfruit' },
+  { text: 'Cherry', value: 'cherry' },
+  { text: 'Cranberry', value: 'cranberry' },
+  { text: 'Durian', value: 'durian' },
+  { text: 'Guava', value: 'guava' },
+  { text: 'Papaya', value: 'papaya' }
+]
+
+export const fruitOptionsFunction = () => Promise.resolve(fruitOptions)
```
Comment (jvanveen) [inline packages/components-library/tests/demo-data/filterMocks.ts:20]: Please don't add type to variable names. Just rename to `fruitOptions`?

File: packages/components-library/tests/unit/filters/ActiveFilters.spec.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  })
+
+  describe('rangeFilter', () => {
+    it('will show less than values', async (done) => {
+      await wrapper.setProps({ value: { range: [null, 2] } })
+      expect(wrapper.vm.activeValues[0].value).toEqual('2 and less')
+      done()
+    })
+
+    it('will show greater values', async (done) => {
```
Comment (jvanveen) [inline packages/components-library/tests/unit/filters/ActiveFilters.spec.ts:85]: What does this mean to a user? (which use-case is tested here?) if-then variable tests should be avoided.

**Review by jvanveen — CHANGES_REQUESTED** [review]

File: packages/components-library/tests/demo-data/filterMocks.ts (hunk 2 above)
Comment (connoratrug) [reply packages/components-library/tests/demo-data/filterMocks.ts:20]: In the general case a would agree, but as this variable is use in the example ( that does not support typescript) it is not clear that fruitOptions is a function ( fruitOptions is a list of fruit). Adding the Function to the name help with both the test and example readability in this case i would argue

File: packages/components-library/package.json [hunk 4]
```diff
@@ -25,6 +25,7 @@
     "@fortawesome/fontawesome-svg-core": "^1.2.30",
     "@fortawesome/free-solid-svg-icons": "^5.14.0",
     "@fortawesome/vue-fontawesome": "^2.0.0",
+    "@molgenis/rsql": "^0.2.5",
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply packages/components-library/package.json:28]: removed

File: packages/components-library/tests/unit/filters/ActiveFilters.spec.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      }
+    )
+  }
+  const startDate = new Date('10/20/2019')
+  const endDate = new Date('10/10/2019')
+  let wrapper: any
+
+  beforeEach(() => {
+    wrapper = mount(ActiveFilters, {
+      propsData: {
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply packages/components-library/tests/unit/filters/ActiveFilters.spec.ts:22]: prefer to keep smaller set to keep test more manageable

File: packages/components-library/tests/unit/filters/ActiveFilters.spec.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+          type: 'date-time-filter'
+        }]
+      }
+    })
+  })
+
+  it('removes filter values on click', () => {
+    wrapper.vm.removeFilter({ key: 'string' })
+    expect(wrapper.emitted().input[0][0]).toEqual({ checkbox: ['red'], datetime: [startDate, endDate], range: [-5, 5], multi: ['blue', 'purple'] })
+    wrapper.vm.removeFilter({ key: 'checkbox', subKey: 'red' })
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply packages/components-library/tests/unit/filters/ActiveFilters.spec.ts]: done

File: packages/components-library/src/components/filters/ActiveFilters.vue (hunk 1 above)
Comment (connoratrug) [reply packages/components-library/src/components/filters/ActiveFilters.vue]: added

File: packages/components-library/tests/unit/filters/ActiveFilters.spec.ts (hunk 3 above)
Comment (connoratrug) [reply packages/components-library/tests/unit/filters/ActiveFilters.spec.ts]: updated test

**Review by jvanveen — APPROVED** [review]

**Pull request merged**